### ISSUE TITLE
Explorer freeze on Windows 2016

### ISSUE THREAD

**Issue by SamKr** [Open-Shell/Open-Shell-Menu#132]
We're using OpenShell StartMenu 4.4.131.0 on a Windows 2016 server, which acts as a terminal server. On multiple occasions and with multiple Windows 2016 instances, we've found that after a while Explorer grinds to a halt and the system becomes unusable until Explorer is restarted.

After eliminating all other possibilities, it would appear that it comes down to OpenShell. I've tried analyzing a dumpfile but haven't come up with anything. 

You can download the dump here: [https://1drv.ms/u/s!ArByg8NWg301ocl-_wsX8IWiceS6jQ ](https://1drv.ms/u/s!ArByg8NWg301ocl-_wsX8IWiceS6jQ).

If you need more information please let me know. Thanks!

**Comment by deasterix** [comment]
Hi,

Exactly same issue here on every single Windows 2016 server. Issue appears randomly and for any user. So far any workaround like spawning every explorer window in a separate process hasn't fixed or improved the issue.

Dump can be found here [https://1drv.ms/u/s!AjdnLV0DXJijge5Ko089Z51jqixESQ](url)

Regarsdd

**Comment by Moopere** [comment]
Are you sure this is not a continuation of a bug that was present with Classic Shell for a long time?

There are several threads relating to an explorer lockup over on the CS forums, one of which is:

http://www.classicshell.net/forum/viewtopic.php?f=12&t=5954

The classic way to tell if its the same bug is to look at the number of threads Explorer is using.  The old bug would cause explorer to lock up when the number of threads passes about 800.  Restarting Explorer.exe fixes the problem until it happens again.

**Comment by deasterix** [comment]
I am pretty sure it is the same bug. However since development of Classic Shell ceased the issue hasn't been fixed since then.

**Comment by XenHat** [comment]
And how sure as we this is not a bug from Explorer itself? I've had similar shell issues with windows 10 enterprise with and without Open-Shell for years now.

**Comment by tehgarra** [comment]
i seem to be having the same issue  with just a regular windows 10 homr installation. worked fine for a while but now explorer just freezes and responds whenever. i've only been using classic shell but it seems to be uninstalled now and i still have the same issue. i only started noticing the problem the last few weeks after i started regularly connecting to the net more 
![update history](https://user-images.githubusercontent.com/15840963/51900283-ebc83d00-2382-11e9-8e0e-e3d8908d5c1a.png)

**Comment by deasterix** [comment]
> 
> 
> And how sure as we this is not a bug from Explorer itself? I've had similar shell issues with windows 10 enterprise with and without Open-Shell for years now.

Hi, we have several hundred users connecting to the RDSH 2016 environment with over a thousand others still waiting to get migrated from 2008R2. Since we simply deactivated the launch of OpenShell for users via the official GPO and users use the Win10 start menu, freezing issues are gone completely while we had dozens of occurences every day while it was activated. It's definitely not Explorer causing the issue but OpenShell you can take that for granted. The problem is always the same >800 threads of Explorer stuck in releasing an exclusive lock. So far we have not been able to reproduce the issue by force, so it's unclear what is triggering the issue. At first we thought it might be users using the thumbnail preview or something similar but even with exaggerated behavior while testing in combination with a hundred explorer windows we couldn't reproduce the issue. We think it has something to do with for ex. notification popups of incoming messages from Outlook (we saw that all users were running Outlook while it happened), which triggers the issue over time or maybe some other taskbar notification like flashing icons.

**Comment by NordicAT** [comment]
Maybe our environment is affected by this issue too. (Windows Server 2016 w. latest CU, Citrix XA 7.15-CU3)
Once the taskbar and the start-button does not react to clicks anymore. Perhaps the desktop-icons are missing too.
If you would be able to launch the task-manager at this time and restart the explorer, everything is fine again.
Is this the same as your issue? Did you find out more in the meantime?

Thank you

**Comment by SamKr** [comment]
I didn't find out anything, and without dev feedback I've had to move on to other solutions.

**Comment by deasterix** [comment]
@NordicAT: This is exactly the same issue. No further occurrence since we deactivated OpenShell. Now we only have to deal with the quirks of the standard start menu - but that's another topic.

**Comment by connexx2** [comment]
Facing the same issue in a Windows 10 LTSB VDI environment right now.  Explorer randomly hangs.  If you run task manager and analyze the wait chain for the process, you can see that all of the stacked up threads are waiting on the Windows Audio service (audiosrv).  Killing and restarting that service will free up Explorer.  Uninstalling OpenShell greatly reduces the frequency, but doesn't solve it completely.  

There's a long thread over at Spiceworks on the issue:  https://community.spiceworks.com/topic/2018699-rds-windows-server-2016-desktop-freeze-black-screen?page=1

It looks pretty clearly like a Microsoft problem and ClassicShell/OpenShell are triggering it with greater frequency.

**Comment by NordicAT** [comment]
I can't confirm, that restarting the Windows Audio Service (audiosrv) fixes the frozen Explorer process for me. I also want to mention, that I did install the latest CU KB4493470 (April 2019) without success. The issue did just happen ...

**Comment by SamKr** [comment]
Thanks for the feedback. We have been experiencing (what we thought was unrelated) issues with audiosrv. Forcing us to monitor and when needed restart the service. We'll remove OpenShell from those servers and see if the frequency of the freezes go down.

**Comment by NordicAT** [comment]
I wanted to ask, if this issue is fixed for you all?
Did you get any further?

I had to remove OpenShell from all our systems. User were frustrated and made me trouble. 
Desktops (start menu and taskbar) were freezing all the time.

Turning off OpenShell immediately stopped the issues. No further occurrence.

I think you got rid of OpenShell too?

**Comment by SamKr** [comment]
Nope not fixed, and with removing OpenShell the problems disappeared. Because of lack of feedback we're considering this abandonware, and have for now moved on to paid alternatives.

**Comment by Moopere** [comment]
Seems to me like this is the same bug we've had since the closing versions of Classic Shell.  Also seems likely its an inherent bug in Explorer that is triggered by Classic/Open Shell.  

Why do I think its inherent in Explorer?  Because since about 1607 or so, definitely by 1709, and using Classic shell 4.3.1 (the last official one) its unlikely this bug will bite as hard as it did with previous versions of Windows (empirically speaking, I've not observed it)

For those using Server 2016, I'd be using classic shell, the original and last version 4.3.1, I note that the author has wedged in a hack on that version which automatically resets Explorer if the problem manifests (>800 explorer threads).   When using Win10 1511 I'd regularly see a pause in the desktop, followed by a reset and then life was back to normal.  I've not seen this behavior now in a long time ... as I say, since probably 1607 or so. 

Never had a problem with CS and Win8/8.1, server 2012R1/R2, classic shell still works fine on those systems.  

I don't run 2016 but have myriad 2012/R2 + 2019 servers are starting to appear now in numbers.  CS 4.3.1 still seems to be the one to use.

**Comment by NordicAT** [comment]
@Moopere :

> Seems to me like this is the same bug we've had since the closing versions of Classic Shell.

Do you mean, that you think the issues started with Classic Shell 4.3.1 and maybe were not there in Classic Shell 4.3.0?

**Comment by ibuprophen1** [comment]
I've read through (and did a bit of research as well) and I've came to the following 2 conclusions... 

1) I can't reproduce this at all. I've tried numerous times (on 3 PC/LPTP's) with different versions of Windows installed and this just wasn't happening for me. 

2) Just as @XenHat had stated on an earlier post here, and on the https://github.com/Open-Shell/Open-Shell-Menu/issues/134#issuecomment-458612243 as well, that this doesn't look like an issue related specifically to the Open Shell software itself. 

Ultimately, I believe that this issue is more likely to be some type of Software Conflict that is either a result of 1 (or more) other softwares that may not be playing well (so to speak) with Open Shell AND/OR the Windows OS having some type of conflict that seems to be more prominent with the Open Shell software. 

I'm not at all invalidating what your experiencing with this issue rather than my personal ability to reproduce it under the same/similar Windows environment myself. 

This issue may need to remain tagged as a Duplicate of the https://github.com/Open-Shell/Open-Shell-Menu/issues/134 that was previously Closed as well. 

I do apologize for what you're going through with this software and I do hope that you are able to locate a solution to your problem. 

I wish you the best of luck! 👍 


~Ibuprophen
